### PR TITLE
(SERVER-1315) Add helper for extracting CA cert from bundle

### PR DESCRIPTION
**Pull request by adrienthebo**
This pull request adds a helper for extracting the first CA certificate from a CA
certificate bundle and verifying that it matches a given public key. This
is meant for certificate chain bundles where we need information from the
leaf CA certificate.

### CODE REVIEW

**Comment by adrienthebo** [comment]
@MosesMendoza @justinstoller we might not wind up using this change, but I thought I'd throw it up for the sake of review and/or posterity.

File: src/clojure/puppetlabs/ssl_utils/core.clj [hunk 1]
```diff
@@ -574,6 +574,15 @@
   (with-open [r (reader pem)]
     (SSLUtils/pemToCerts r)))
 
+(schema/defn ^:always-validate pem->ca-cert :- X509Certificate
+  "Given a CA certificate bundle and public key, extract the first certificate and
```
Comment (justinstoller) [inline src/clojure/puppetlabs/ssl_utils/core.clj]: We expect our certificate to be the first certificate in the bundle. Doesn't that make it a chain? If we expect an unordered bundle here should we have a different implementation of `pemToCaCert`?

Comment (adrienthebo) [inline src/clojure/puppetlabs/ssl_utils/core.clj]: Fair point, I'll rename things accordingly.

**Comment by justinstoller** [comment]
I don't think you need to do it asap since we don't know what our
requirements for this will end up being, but I think this work is generally
useful, well done, and something we'll very likely use without much
adjustment in the end.   :+1:

**Comment by MosesMendoza** [comment]
awesome

**Pull request merged**